tsl/src/reorder.c

```diff
@@ -148,12 +149,53 @@ tsl_move_chunk(PG_FUNCTION_ARGS)
 				 errmsg("valid chunk, destination_tablespace, and index_destination_tablespaces "
 						"are required")));
 
-	reorder_chunk(chunk_id,
-				  index_id,
-				  verbose,
-				  wait_id,
-				  destination_tablespace,
-				  index_destination_tablespace);
+	chunk = ts_chunk_get_by_relid(chunk_id, false);
+
+	if (NULL == chunk)
+		ereport(ERROR,
+				(errcode(ERRCODE_INVALID_PARAMETER_VALUE),
+				 errmsg("\"%s\" is not a chunk", get_rel_name(chunk_id))));
+
+	if (ts_chunk_contains_compressed_data(chunk))
+	{
+		Chunk *chunk_parent = ts_chunk_get_compressed_chunk_parent(chunk);
+
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot directly move internal compression data"),
+				 errdetail("Chunk \"%s\" contains compressed data for chunk \"%s\" and cannot be "
+						   "moved directly.",
+						   get_rel_name(chunk_id),
+						   get_rel_name(chunk_parent->table_id)),
+				 errhint("Moving chunk \"%s\" will also move the compressed data.",
+						 get_rel_name(chunk_parent->table_id))));
+	}
+
+	/* If chunk is compressed move it by altering tablespace on both chunks */
+	if (OidIsValid(chunk->fd.compressed_chunk_id))
+	{
+		Chunk *compressed_chunk = ts_chunk_get_by_id(chunk->fd.compressed_chunk_id, true);
+		AlterTableCmd cmd = { .type = T_AlterTableCmd,
+							  .subtype = AT_SetTableSpace,
+							  .name = get_tablespace_name(destination_tablespace) };
+
+		ereport(NOTICE,
```

Comment:
not a fan of this being NOTICE, we are already way too spammy with NOTICES imo i think this should be DEBUG1

Reply:
@erimatnor, @gayyappan are you ok with it?

Reply:
This should be conditional message.
Maybe reword it slightly as well.

if ( IsValidOid( index_id) ) { 
      elog( NOTICE, "Ignoring index parameter. Chunk will not be reordered as it has compressed data");
}

Reply:
@gayyappan updated